[CT-1281] Add possibility to use environment variables for the credentials

vvnkr opened the pull request:
Fixes [CT-1281](https://datacamp.atlassian.net/browse/CT-1281)

- Allow us to overwrite the DB host by using an environment variable (e.g. `DBCONNECT_ENDPOINT`)

Commit:
feat: add possibility to overwrite credentials with env variables

hermansje commented on R/creds.R:
```diff
@@ -22,11 +22,13 @@ get_creds <- function(dbname = "main-app", cache = FALSE, cache_folder = "~/.dat
   return(creds)
 }
 
+fields <- c(user = "user", password = "password", host = "endpoint", port = "port", drv = "type")
```
```suggestion
cred_fields <- c(user = "user", password = "password", host = "endpoint", port = "port", drv = "type")
```

hermansje reviewed (approved):
LGTM.

Having little context on this repo, it took a while to figure out that `drv` stands for 'driver', but that's besides the point of this PR.

hermansje commented:
@ramnathv your review will still be very useful since you know this area better than me! (do you know why the tests are failing?)

ramnathv commented on R/creds.R:
```diff
@@ -59,6 +61,30 @@ fetch_creds <- function(dbname = "main-app", ...) {
+  if (env_variable_value == "") {
+    return(value)
+  }
+
```
This can be simplified to:

```r
env_variable_value <- Sys.getenv(env_variable, value)
```

ramnathv commented on R/creds.R:
```diff
@@ -59,6 +61,30 @@ fetch_creds <- function(dbname = "main-app", ...) {
+  env_variable <- tryCatch(
+    paste0("DBCONNECT_", toupper(fields[[key]])),
+    error=function(...) NULL
+  )
```
I am not clear why a `tryCatch` is needed here. Is it because the key might not exist in `fields`?

vvnkr replied:
Yes indeed

ramnathv reviewed (approved):
I made a few suggestions. I am approving it so it is not a bottleneck for merging.

ramnathv commented:
> @ramnathv your review will still be very useful since you know this area better than me! (do you know why the tests are failing?)

@hermansje I think the tests are failing due to an expired `GITHUB_PAT`. Most likely, we revoked the token after the CircleCI breach but did NOT replace it with a new one here. https://app.travis-ci.com/github/datacamp/dbconnectr/jobs/538446332.

Commit:
refactor: change naming and reduce complexity

- Rename `fields` to `cred_fields` in file scope
- Use default value in `Sys.getenv` to reduce complexity of `get_env_overwrite`
